tests/common/e2e_test.go

```diff
@@ -17,8 +17,59 @@
 
 package common
 
-import "go.etcd.io/etcd/tests/v3/framework"
+import (
+	"go.etcd.io/etcd/client/pkg/v3/fileutil"
+	"go.etcd.io/etcd/tests/v3/framework"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	"go.etcd.io/etcd/tests/v3/framework/e2e"
+)
 
 func init() {
 	testRunner = framework.E2eTestRunner
+	clusterTestCases = e2eClusterTestCases
+}
+
+func e2eClusterTestCases() []testCase {
+	tcs := []testCase{
+		{
+			name:   "NoTLS",
+			config: config.ClusterConfig{ClusterSize: 1},
+		},
+		{
+			name:   "PeerTLS",
+			config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.ManualTLS},
+		},
+		{
+			name:   "PeerAutoTLS",
+			config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.AutoTLS},
+		},
+		{
+			name:   "ClientTLS",
+			config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.ManualTLS},
+		},
+		{
+			name:   "ClientAutoTLS",
+			config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.AutoTLS},
+		},
+	}
```

Comment:
These cases are duplicated to the items in `integrationClusterTestCases`. Have you considered to define a `commonClusterTestCases`?

Reply:
I considered it, but found I don't think we should do a premature refactor and jump on every code duplication.

Reply:
Although I don't entirely agree this, it isn't a big deal.